### PR TITLE
Feature - 5878 - pool creation team selection

### LINKED ISSUE #5878: ♻️ Update pool creation with teams
## ♻️ Debt/Refactor

Need to update pool creation to use teams properly. 

## 🕵️ Details

As a placeholder, pool creation automatically connects to the DCM team. This will require updating to connecting a pool to the appropriate team.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/0021af526932cde208fc6dd42da9d31583c0f585/apps/web/src/pages/Pools/CreatePoolPage/CreatePoolPage.tsx#L254-L260

## 🙋‍♀️ Proposed Solution

This can be a bit tricky as a user could be a pool operator for more than one team, in which case,
 `me -> roleAssignments -> poolOperator -> team.id`  
wouldn't be a safe choice.

We don't have a figma design for this, so add the new select input to the form in a way that looks consistent.

## ✅ Acceptance Criteria

- [ ] Instead of automatically adding the dcm team id, Team is selected in a dropdown element on the Create Team page
- [ ] Dropdown is populated with the teams the user belongs to as a pool operator

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #5878 

## 👋 Introduction

Incorporates team selection in the pool creation process. 

## 🕵️ Details

Just a matter of updating queries and passing in a teams array to the form. 
Overhauled a few things too, like the unused genericJobTitles collection. 

## 🧪 Testing

1. play around with pool creation, try to think of edge cases
2. confirm you can create pools and they attach to teams, and your teams are appearing correctly

a query to check your teams and their associated pools

```
query getMyPools {
  me {
    roleAssignments {
      team {
        id 
        name
        pools {
          id 
          name {
            en
          }
        }
      }
    }
  }
}
```

## 📸 Screenshot

image

### CODE REVIEW

**Comment by vd1992** [comment]
@Jerryescandon 
any opinions on the wording of the new stuff, this bit in particular
opinions as in concerns

image

**Review by petertgiles — APPROVED** [review]
Looks great!  :+1:

File: apps/web/src/lang/fr.json [hunk 1]
```diff
@@ -6786,10 +6786,6 @@
     "defaultMessage": "Rôles des membres",
     "description": "Label for the input to add roles to a user's team membership"
   },
-  "eW7I5E": {
-    "defaultMessage": "Sélectionner les rôles...",
-    "description": "Placeholder text for role selection input"
-  },
```
Comment (vd1992) [inline apps/web/src/lang/fr.json:6792]: @petertgiles 
translation added in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5904/commits/268e08b374ad8589dee53eac5e916ee95ac5c5a6
this deletion appears to have been a duplicate that was removed by one of the npm intl commands, translation check is passing

Comment (vd1992) [reply apps/web/src/lang/fr.json:6792]: I ran a few intl things to try so that's why I am not sure which one removed it 😅

Comment (petertgiles) [reply apps/web/src/lang/fr.json:6792]: :+1:

**Pull request merged**